backend/cfg/cfg.ml

```diff
@@ -244,23 +244,28 @@ let dump_call ppf = function
     | Direct { func_symbol : string; _ } ->
       Format.fprintf ppf "direct %s" func_symbol)
 
-let dump_basic ppf i =
+let dump_basic ppf (i : basic instruction) =
   let open Format in
+  if i.stack_offset > 0 then fprintf ppf "[T%d] " i.stack_offset;
   fprintf ppf "%d: " i.id;
-  match i.desc with
+  if Array.length i.res > 0 then fprintf ppf "%a := " Printmach.regs i.res;
+  (match i.desc with
   | Op op -> dump_op ppf op
   | Call call ->
     fprintf ppf "Call ";
     dump_call ppf call
   | Reloadretaddr -> fprintf ppf "Reloadretaddr"
   | Pushtrap { lbl_handler } -> fprintf ppf "Pushtrap handler=%d" lbl_handler
   | Poptrap -> fprintf ppf "Poptrap"
-  | Prologue -> fprintf ppf "Prologue"
+  | Prologue -> fprintf ppf "Prologue");
+  fprintf ppf " %a" Printmach.regs i.arg
 
-let dump_terminator ppf ?(sep = "\n") ti =
+let dump_terminator ppf ?(sep = "\n") (ti : terminator instruction) =
```

Comment:
(We could factor out the parts which
are common to "basic" and "terminator"
instructions.)

Reply:
done